cmd/bootstrap/cmd/key.go

```diff
@@ -9,78 +9,120 @@ import (
+
+	// required flags
+	keyCmd.Flags().StringVar(&flagRole, "role", "", "node role (can be \"collection\", \"consensus\", \"execution\", \"verification\" or \"access\")")
+	_ = keyCmd.MarkFlagRequired("role")
+	keyCmd.Flags().StringVar(&flagAddress, "address", "", "network address")
+	_ = keyCmd.MarkFlagRequired("address")
+
+	keyCmd.Flags().BytesHexVar(&flagNetworkSeed, "networking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded networking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagStakingSeed, "staking-seed", generateRandomSeed(), fmt.Sprintf("hex encoded staking seed (min %v bytes)", minSeedBytes))
+	keyCmd.Flags().BytesHexVar(&flagMachineSeed, "machine-seed", generateRandomSeed(), fmt.Sprintf("hex encoded machine account seed (min %v bytes)", minSeedBytes))
+}
+
 // keyCmdRun generate the node staking key, networking key and node information
 func keyCmdRun(_ *cobra.Command, _ []string) {
+
+	// TODO: We need NodeID of the node to continue checking if the node-info.priv.json exists
+	// TODO: Add paths to log messages for clarity
+	exists, err := pathExists(fmt.Sprintf(model.PathNodeInfoPriv, flow.Identifier{}))
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not check if node-info.priv.json exists")
+	}
+	if exists {
+		log.Warn().Msg("node-info.priv.json exists, please delete and try again if you are tying to generate new keys")
+		return
+	}
+
 	// validate inputs
 	role := validateRole(flagRole)
 	validateAddressFormat(flagAddress)
-	networkSeed := validateSeed(flagNetworkSeed)
-	stakingSeed := validateSeed(flagStakingSeed)
 
-	log.Debug().Msg("will generate networking key")
-	networkKeys, err := run.GenerateNetworkingKeys(1, [][]byte{networkSeed})
+	// generate all three keys
+	networkKey, stakingKey, machineKey, err := generateKeys()
 	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate networking key")
+		log.Fatal().Err(err).Msg("could not generate keys")
 	}
-	log.Info().Msg("generated networking key")
 
-	log.Debug().Msg("will generate staking key")
-	stakingKeys, err := run.GenerateStakingKeys(1, [][]byte{stakingSeed})
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot generate staking key")
-	}
-	log.Info().Msg("generated staking key")
+	// construct NodeMachineAccountInfo struct to write
+	machineAccountInfo := assembleNodeMachineAccountInfo(machineKey)
 
 	log.Debug().Str("address", flagAddress).Msg("assembling node information")
 	conf := model.NodeConfig{
 		Role:    role,
 		Address: flagAddress,
 		Stake:   0,
 	}
-	nodeInfo := assembleNodeInfo(conf, networkKeys[0], stakingKeys[0])
+	nodeInfo := assembleNodeInfo(conf, networkKey, stakingKey)
 
 	// retrieve private representation of the node
 	private, err := nodeInfo.Private()
 	if err != nil {
 		log.Fatal().Err(err).Msg("could not access private keys")
 	}
 
+	// write files
 	writeText(model.PathNodeID, []byte(nodeInfo.NodeID.String()))
 	writeJSON(fmt.Sprintf(model.PathNodeInfoPriv, nodeInfo.NodeID), private)
```

Comment:
We can sanity check that the file doesn't exist here before writing.

Reply:
Would this be needed as we create new keys which will always return as `exists = false`?

Reply:
You're right, probably not needed here. But definitely needed for the command which only produces the machine account key